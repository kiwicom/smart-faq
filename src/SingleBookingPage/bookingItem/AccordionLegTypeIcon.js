// @flow

import * as React from 'react';
import { createFragmentContainer, graphql } from 'react-relay';
import css from 'styled-jsx/css';
import AirplaneDown from '@kiwicom/orbit-components/lib/icons/AirplaneDown';
import Train from '@kiwicom/orbit-components/lib/icons/Train';
import Bus from '@kiwicom/orbit-components/lib/icons/Bus';

import type { AccordionLegTypeIcon_leg } from './__generated__/AccordionLegTypeIcon_leg.graphql';
import bookingLegTypes from '../../common/booking/bookingLegTypes';

const IconStyle = css`
  div.legTypeIcon {
    position: absolute;
    left: -13px;
    top: 40px;
    background: #fff;
  }
`;

type IconProps = {
  leg: AccordionLegTypeIcon_leg,
};

const LegTypeIcon = (props: IconProps) => {
  const type = props.leg.type;
  let icon;
  switch (type) {
    case bookingLegTypes.AIRCRAFT:
      icon = <AirplaneDown customColor="#adb9c5" size="medium" />;
      break;

    case bookingLegTypes.TRAIN:
      icon = <Train customColor="#00a991" size="medium" />;
      break;

    case bookingLegTypes.BUS:
      icon = <Bus customColor="#00a991" size="medium" />;
      break;

    default:
      icon = null;
  }
  return (
    <React.Fragment>
      {icon ? (
        <div className="legTypeIcon">
          {icon}
          <style jsx>{IconStyle}</style>
        </div>
      ) : null}
    </React.Fragment>
  );
};

export const PureLegTypeIcon = LegTypeIcon;

export default createFragmentContainer(
  LegTypeIcon,
  graphql`
    fragment AccordionLegTypeIcon_leg on Leg {
      type
    }
  `,
);
