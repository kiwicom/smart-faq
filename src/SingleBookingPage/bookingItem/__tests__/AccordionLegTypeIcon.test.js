// @flow
import * as React from 'react';
import { shallow } from 'enzyme';

import { PureLegTypeIcon } from '../AccordionLegTypeIcon';

describe('LegTypeIcon', () => {
  it('should render aircraft icon', () => {
    const leg = {
      type: 'AIRCRAFT',
    };
    const wrapper = shallow(<PureLegTypeIcon leg={leg} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render bus icon', () => {
    const leg = {
      type: 'BUS',
    };
    const wrapper = shallow(<PureLegTypeIcon leg={leg} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render train icon', () => {
    const leg = {
      type: 'TRAIN',
    };
    const wrapper = shallow(<PureLegTypeIcon leg={leg} />);
    expect(wrapper).toMatchSnapshot();
  });
});
