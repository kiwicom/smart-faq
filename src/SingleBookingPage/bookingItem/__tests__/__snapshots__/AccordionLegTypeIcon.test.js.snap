// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegTypeIcon should render aircraft icon 1`] = `
<React.Fragment>
  <div
    className="jsx-885109022 legTypeIcon"
  >
    <AirplaneDown
      customColor="#adb9c5"
      height="12"
    />
    <JSXStyle
      css="div.legTypeIcon.jsx-885109022{position:absolute;left:-13px;top:40px;background:#fff;}"
      styleId="885109022"
    />
  </div>
</React.Fragment>
`;

exports[`LegTypeIcon should render bus icon 1`] = `
<React.Fragment>
  <div
    className="jsx-885109022 legTypeIcon"
  >
    <Bus
      customColor="#00a991"
      height="12"
    />
    <JSXStyle
      css="div.legTypeIcon.jsx-885109022{position:absolute;left:-13px;top:40px;background:#fff;}"
      styleId="885109022"
    />
  </div>
</React.Fragment>
`;

exports[`LegTypeIcon should render train icon 1`] = `
<React.Fragment>
  <div
    className="jsx-885109022 legTypeIcon"
  >
    <Train
      customColor="#00a991"
      height="12"
    />
    <JSXStyle
      css="div.legTypeIcon.jsx-885109022{position:absolute;left:-13px;top:40px;background:#fff;}"
      styleId="885109022"
    />
  </div>
</React.Fragment>
`;
